src/Features/Core/Portable/Diagnostics/IBuiltInAnalyzer.cs

```diff
@@ -8,6 +8,20 @@ namespace Microsoft.CodeAnalysis.Diagnostics
     /// </summary>
     internal interface IBuiltInAnalyzer
     {
+        /// <summary>
+        /// This category will be used to run analyzer more efficiently by restricting scope of analysis
+        /// </summary>
         DiagnosticAnalyzerCategory GetAnalyzerCategory();
+
+        /// <summary>
+        /// This indicates whether this builtin analyzer must run in proc or can be run on remote host such as service hub.
+        /// 
+        /// if the diagnostic analyzer can run in command line as it is, then it should be able to run in remote host. 
+        /// otherwise, it won't unless diagnostic analyzer author make changes in remote host to provide whatever missing
+        /// data command line build doesn't provide such as workspace options/services/MEF and etc.
+        /// 
+        /// at this moment, remote host provide same context as command line build and only that context
+        /// </summary>
+        bool MustRunInProc { get; }
```

Comment:
LGTM, though I would still prefer a more appropriate property name. IDE folks can decide what works for them..